src/Compilers/CSharp/Portable/FlowAnalysis/FlowAnalysisPass.cs

```diff
@@ -87,6 +101,28 @@ internal class FlowAnalysisPass
             return block;
         }
 
+        private static BoundBlock PrependImplicitInitializations(BoundBlock body, MethodSymbol method, ImmutableArray<FieldSymbol> implicitlyInitializedFields, TypeCompilationState compilationState, BindingDiagnosticBag diagnostics)
+        {
+            Debug.Assert(method.MethodKind == MethodKind.Constructor);
+            Debug.Assert(method.ContainingType.IsStructType());
+
+            var syntax = body.Syntax;
+            var F = new SyntheticBoundNodeFactory(method, syntax, compilationState, diagnostics);
+
+            var builder = ArrayBuilder<BoundStatement>.GetInstance(implicitlyInitializedFields.Length + 1);
+            foreach (var field in implicitlyInitializedFields)
+            {
+                builder.Add(new BoundExpressionStatement(
```

Comment:
Do we need a use site diagnostic here? see [`CodeGenTupleTests.CustomValueTuple_StructWithConstructor`](https://github.dev/dotnet/roslyn/blob/ffc61252d3cb51c7a56ee4ae2a6adfcfcd1aa1f7/src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs#L13375)

It might make more sense to give a "Missing compiler required member" on the tuple declaration which is missing the well-known fields.

Reply:
I'm leaning toward allowing the current behavior here. Declaring your own ValueTuple is a pretty rare thing to do and I wasn't able to find a decent way to introduce the use site diagnostics here without adding redundant use site diagnostics in other scenarios.